fw/http_frame.c

```diff
@@ -895,26 +893,21 @@ static void
 tfw_h2_closed_streams_shrink(TfwH2Ctx *ctx)
 {
 	TfwStream *cur;
-	unsigned int max_streams = ctx->lsettings.max_streams;
 	TfwStreamQueue *closed_streams = &ctx->closed_streams;
 
 	T_DBG3("%s: ctx [%p] max_streams %u\n", __func__, ctx, max_streams);
 
 	while (1) {
 		spin_lock(&ctx->lock);
 
-		if (closed_streams->num <= TFW_MAX_CLOSED_STREAMS
-		    || (max_streams == ctx->streams_num
-			&& closed_streams->num))
```

Comment:
IIUC this condition need to not delete streams if we have maximum streams and some streams in closed queue. These streams will be deleted on closing connection. Is there any reason to remove it?

Reply:
Yes, fixed during discussion. Add comment to https://github.com/tempesta-tech/tempesta/issues/1346 to don't forget to change this condition later